### PR TITLE
Added a Settings Command.

### PR DESCRIPTION
**Pull request by ALEXWARELLC**
Added a settings command which will print out the configuration information with colors.

Green - True
Red - False
Yellow - Other

### CODE REVIEW

**Comment by Jesus-QC** [comment]
First of all, you are using the same code for each setting to print in the desired color the value, make that a method.

Second of all, you could make a method inside the config class and return that value.

And third, there is no need to color anything, just return config::ToString()

**Comment by zabszk** [comment]
Hi,

First of all conversion of the config to string already exists here: https://github.com/northwood-studios/LocalAdmin-V2/blob/593f190c71bfbde66fb23e2c4d73ac7fa116571e/IO/Config.cs#L234-L280

Secondly LocalAdmin is a multithreaded application. Printing output of this command line by line (using separate Write/WriteLine commands) will sooner or later cause an issue where two different Write/WriteLines run concurrently on different threads. That will result in output from this command being mixed with a different text (printed on a different thread). You additionally set foreground color, which will make the situation even worse.

And
https://github.com/northwood-studios/LocalAdmin-V2/blob/593f190c71bfbde66fb23e2c4d73ac7fa116571e/Core/LocalAdmin.cs#L20-L29

**Comment by ALEXWARELLC** [comment]
Honestly, I just wanted to get it done. And I had it in a method but  is would make the code a bit longer than already. But yes, I'm working on the method part.

Secondly, I understand the mixing issue and I'm trying to figure out the best way to implement it without that issue. And I am trying to find a better solution for it.

**Comment by ALEXWARELLC** [comment]
> First of all, you are using the same code for each setting to print in the desired color the value, make that a method.
> 
> Second of all, you could make a method inside the config class and return that value.
> 
> And third, there is no need to color anything, just return config::ToString()
> 
> Anyways that is my personal feedback, wait for Zabszk to see what he says.

That might be the better solution. I'll try to change it to that now.

**Comment by zabszk** [comment]
> It should be an easy command smt like this:
> 
> ```cs
> internal override void Execute(string[] arguments) => ConsoleUtil.WriteLine(Configuration.ToString());
> ```
> 
> Again, personal feedback.
> 
> Also about the command name I would maybe change it to smt more LA related so it doesnt break any type of plugins that may be using that.
> 
> But zabszk may have their opinion about that.

Plugins don't load into LA, so it's not an issue. These commands are accessible only from the LA.

File: Commands/LASettingsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using LocalAdmin.V2.IO;
+using LocalAdmin.V2.Commands.Meta;
+using LocalAdmin.V2.Core;
+using System;
+
+namespace LocalAdmin.V2.Commands;
+
+internal sealed class LASettingsCommand : CommandBase
+{
+    Config Settings;
```
Comment (zabszk) [inline Commands/LASettingsCommand.cs:10]: Unused variable?

File: Commands/LASettingsCommand.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    public LASettingsCommand() : base("settings") { }
```
Comment (zabszk) [inline Commands/LASettingsCommand.cs:11]: You might want to rename the command itself, not the class.

As I explained to @Jesus-QC here https://github.com/northwood-studios/LocalAdmin-V2/pull/72#issuecomment-1676370734 having such command prevents executing "settings" command on the server console.

A similar command is named "lacfg". You might want to rename this to "lasettings" or actually merge these two commands into one, so "lacfg" prints both the path and the configuration.

File: Core/Program.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```
Comment (zabszk) [inline Core/Program.cs:1]: What's the change here?

**Review by zabszk — CHANGES_REQUESTED** [review]
Additionally keep in mind that "help" command isn't automatically generated. You need to add the command there as well.

File: Core/Program.cs (hunk 3 above)
Comment (ALEXWARELLC) [reply Core/Program.cs:1]: No idea, but it does absolutely nothing. I never touched that file.

File: Commands/LASettingsCommand.cs (hunk 2 above)
Comment (ALEXWARELLC) [reply Commands/LASettingsCommand.cs:11]: OK, I'll merge lacfg with settings and make it the main way to see configuration settings.

**Comment by Jesus-QC** [comment]
> > It should be an easy command smt like this:
> > 
> > ```cs
> > internal override void Execute(string[] arguments) => ConsoleUtil.WriteLine(Configuration.ToString());
> > ```
> 
> Yes. Something like this should be enough.
> 
> > Again, personal feedback.
> > 
> > Also about the command name I would maybe change it to smt more LA related so it doesnt break any type of plugins that may be using that.
> > 
> > But zabszk may have their opinion about that.
> 
> While plugins don't interact with LA, having a LA command named `settings` means that you can't execute such command as server console. The command already existing for showing config path is named "lacfg" to clearly indicate it's an LA config.

Yeah, that was what I was afraid of, server console commands.

**Comment by ALEXWARELLC** [comment]
> > > It should be an easy command smt like this:
> > > ```cs
> > > internal override void Execute(string[] arguments) => ConsoleUtil.WriteLine(Configuration.ToString());
> > > ```
> > 
> > 
> > Yes. Something like this should be enough.
> > > Again, personal feedback.
> > > Also about the command name I would maybe change it to smt more LA related so it doesnt break any type of plugins that may be using that.
> > > But zabszk may have their opinion about that.
> > 
> > 
> > While plugins don't interact with LA, having a LA command named `settings` means that you can't execute such command as server console. The command already existing for showing config path is named "lacfg" to clearly indicate it's an LA config.
> 
> Yeah, that was what I was afraid of, server console commands.
> 
> With breaking I meant being overrided.

It still gives the same idea. But I get where your trying to point.

File: Commands/LaCfgCommand.cs [hunk 4]
```diff
@@ -11,5 +11,6 @@ internal sealed class LaCfgCommand : CommandBase
     internal override void Execute(string[] arguments)
     {
         ConsoleUtil.WriteLine($"Current LocalAdmin config file path is: {Core.LocalAdmin.CurrentConfigPath ?? "(null)"}", ConsoleColor.DarkGreen);
+        ConsoleUtil.WriteLine($"Current LocalAdmin Configuration: \n{LocalAdmin.V2.Core.LocalAdmin.Configuration.ToString()}");
```
Comment (zabszk) [inline Commands/LaCfgCommand.cs:14]:
```suggestion
        ConsoleUtil.WriteLine($"Current LocalAdmin Configuration:\n{LocalAdmin.V2.Core.LocalAdmin.Configuration.ToString()}");
```

File: Core/Program.cs (hunk 3 above)
Comment (zabszk) [reply Core/Program.cs:1]: If it's an unknown change it shouldn't be commited. Before commiting you should check all changes and commit only the intended ones.

Actually it seems that `using System;` is not even needed here, so you might want to try removing it completely.

**Review by zabszk — CHANGES_REQUESTED** [review]

File: Commands/LaCfgCommand.cs [hunk 5]
```diff
@@ -11,5 +11,6 @@ internal sealed class LaCfgCommand : CommandBase
     internal override void Execute(string[] arguments)
     {
         ConsoleUtil.WriteLine($"Current LocalAdmin config file path is: {Core.LocalAdmin.CurrentConfigPath ?? "(null)"}", ConsoleColor.DarkGreen);
+        ConsoleUtil.WriteLine($"Current LocalAdmin Configuration:\n{LocalAdmin.V2.Core.LocalAdmin.Configuration.ToString()}");
```
Comment (zabszk) [inline Commands/LaCfgCommand.cs:14]:
```suggestion
        ConsoleUtil.WriteLine($"Current LocalAdmin Configuration:{Environment.NewLine}{LocalAdmin.V2.Core.LocalAdmin.Configuration!.ToString()}");
```

**Review by zabszk — CHANGES_REQUESTED** [review]

**Comment by zabszk** [comment]
The code looks good. Can you update the command description in the "help" command?

**Comment by ALEXWARELLC** [comment]
> The code looks good. Can you update the command description in the "help" command?

Yes, I can do that rn.

**Comment by ALEXWARELLC** [comment]
Feel free to make suggested cahnges.

**Review by zabszk — APPROVED** [review]

**Pull request merged**